api/tests/unit/domain/usecases/add-certification-candidate-to-session_test.js

```diff
@@ -31,6 +33,47 @@ describe('Unit | UseCase | add-certification-candidate-to-session', function ()
     certificationCpfCityRepository = Symbol('certificationCpfCityRepository');
   });
 
+  context('when the session is finalized', function () {
+    it('should throw an CertificationCandidateOnFinalizedSessionError', async function () {
+      // given
+      sessionRepository.isSco.resolves(false);
+      const complementaryCertifications = [
+        domainBuilder.buildComplementaryCertification(),
+        domainBuilder.buildComplementaryCertification(),
+      ];
+      const certificationCandidate = domainBuilder.buildCertificationCandidate.pro({
+        sessionId: null,
+        complementaryCertifications,
+      });
+      const cpfBirthInformationValidation = CpfBirthInformationValidation.success({
+        birthCountry: 'COUNTRY',
+        birthINSEECode: 'INSEE_CODE',
+        birthPostalCode: null,
+        birthCity: 'CITY',
+      });
+      certificationCandidateRepository.findBySessionIdAndPersonalInfo.resolves([]);
+      certificationCpfService.getBirthInformation.resolves(cpfBirthInformationValidation);
+      certificationCandidateRepository.saveInSession.resolves();
```

Comment:
Ces methodes ne devraient pas etre appelé. On peut ne pas mettre les resolves pour alleger la comprehension du test. Pour aller plus loin on peut s'assurer qu'elle ne sont pas appelés 'not.to.have.been.called'